src/arrow2/reader_async.rs

```diff
@@ -117,18 +92,50 @@ pub async fn read_row_group(
     )
     .await?;
 
+    let deserializer = RowGroupDeserializer::new(column_chunks, row_group_meta.num_rows(), None);
+    Ok(deserializer)
+}
+
+pub async fn read_row_group(
+    url: String,
+    // content_length: Option<usize>,
+    row_group_meta: &RowGroupMetaData,
+    arrow_schema: &Schema,
+    chunk_fn: impl Fn(Chunk<Box<dyn Array>>) -> Chunk<Box<dyn Array>>,
+) -> Result<Vec<u8>> {
+    let deserializer = _read_row_group(url, row_group_meta, arrow_schema).await?;
     // Create IPC writer
     let mut output_file = Vec::new();
     let options = IPCWriteOptions { compression: None };
     let mut writer = IPCStreamWriter::new(&mut output_file, options);
     writer.start(arrow_schema, None)?;
-
-    let deserializer = RowGroupDeserializer::new(column_chunks, row_group_meta.num_rows(), None);
     for maybe_chunk in deserializer {
         let chunk = chunk_fn(maybe_chunk?);
         writer.write(&chunk, None)?;
     }
-
-    writer.finish()?;
     Ok(output_file)
 }
+
+pub async fn read_record_batch_stream(
+    url: String,
+) -> Result<impl futures::Stream<Item = super::ffi::FFIArrowRecordBatch>> {
+    use async_stream::stream;
+    let inner_stream = stream! {
```

Comment:
woah! That's cool! I didn't even know `yield` was a keyword

Reply:
The cake (`yield`) is a lie with this one - that stream! macro slightly extends the grammar (apparently it's on it's way to the language - generator syntax seems to be one of the harder ones to add to a language).

Reply:
Ah I see it's a proc macro so it can make up new keywords!